server/users.ts

```diff
@@ -356,6 +361,9 @@ export class User extends Chat.MessageContext {
 	lastDisconnected: number;
 	lastConnected: number;
 	foodfight?: {generatedTeam: string[], dish: string, ingredients: string[], timestamp: number};
+	// We want this to be null on creation so that the database can be queried only once per user object.
+	friends!: Set<string>;
+	friendRequests!: {sent: Set<string>, received: Set<string>} | null;
```

Comment:
You should really only do this if these variables are set in the constructor, which they don't appear to be.

Reply:
No, but they're set immediately after login in the loginfilter.

Reply:
That's not good enough; it doesn't apply to guests, and the login process can take hundreds of milliseconds anyway. The `!` assertion is only safe if you set it in the same tick (i.e. you don't wait for any Promises or callbacks before setting it).

Reply:
I would recommend `friends: Set<string> | null`, and setting it to `null` in the constructor.

Reply:
Alright.

Reply:
Done.

Reply:
Added bonus, this is hotpatchable with no monkeypatch now!